src/client/java/daos-java/src/main/native/io_daos_dfs_DaosFsClient.c

```diff
@@ -778,6 +778,56 @@ Java_io_daos_dfs_DaosFsClient_dfsRelease(JNIEnv *env,
 	}
 }
 
+/**
+ * allocate and initialize dfs description.
+ *
+ * \param[in]	env		JNI environment
+ * \param[in]	clientClass	DaosFsClient class
+ * \param[in]	objId		pointer to fs object
+ *
+ * \return  pointer address of dfsDesc
+ */
+JNIEXPORT jlong JNICALL
+Java_io_daos_dfs_DaosFsClient_allocateDfsDesc(JNIEnv *env, jclass clientClass,
+        jlong descBufAddress)
+{
+    uint64_t value64;
+    uint16_t value16;
+    char *desc_buffer = (char *)descBufAddress;
+    dfs_desc_t *desc = (dfs_desc_t *)malloc(
+    		sizeof(dfs_desc_t));
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs